server.js

```diff
@@ -45,53 +71,94 @@ function runClient(client) {
         socket.emit('upvote', answers[upvotedId][upvotedNumber]);
       }
     }
+
+    // clearing answers
+    if (answersInQueuePerIP[ipAddress]) {
+      answersInQueuePerIP[ipAddress] -= answers[client.id] ? answers[client.id].length : 0;
+    }
     delete answers[client.id];
     socket.emit('remove answers', client.id);
+
+    // clearing questions
+    if (questions[client.id] && questionsInQueuePerIP[ipAddress]) {
+      questionsInQueuePerIP[ipAddress]--;
+    }
+    if (currentQuestion.id === client.id && questionsInQueuePerIP[ipAddress]) {
+      questionsInQueuePerIP[ipAddress]--;
+    }
+    delete questions[client.id];
+
+    // clearing user
     delete clients[client.id];
     socket.emit('clients online', Object.keys(clients).length);
+
+    // clearing ip-wide data
+    if (answersInQueuePerIP[ipAddress] === 0) {
+      delete answersInQueuePerIP[ipAddress];
+    }
+    if (questionsInQueuePerIP[ipAddress] === 0) {
+      delete questionsInQueuePerIP[ipAddress];
+    }
+
+    log('clearing on disconnect');
   });
 
   client.on('submit question', function(data) {
-    data = data.replace(/</g, "&lt;").replace(/>/g, "&gt;");
-    console.log('question received: ' + data);
+    data = escapeHtml(data);
```

Comment:
Implementing the escapeHtml NPM module. See the Repo for it here: https://github.com/component/escape-html
Similar to your code, just a bit more all-encompassing.